Working Trajopt

ommmid opened the pull request:
- The tesseract env and kin are not used at all. They are replaced with what we have in planning_scene, for that I copied and adjusted the needed files from trajopt_ros, i.e., problem_description and kinematic_terms

- The constraint from planning interface request is converted to the constraint we need in trajopt and that is only for joint state constraint for now.

- No response from tesseract is used. planning interface response is created and fed internally. 

Problems:
- The start state for trajopt is not set, that is probably why I get a solution that has the same joint values for all timesteps except the last timestep 
- start_fixed is set to false. if it is true it means: it constraints the joint values to be their initialized ones but because the start state is not set, then it creates problem. I am not sure about this.
- Please ignore the followings:
There are few struct and classes that are not used
There are headers that are included in different files that might not be used.
Also in CMakeLists, there might be extra or unused packages.

@davetcoleman @BryceStevenWilley @henningkayser @felixvd @j-petit

Commit:
added solve from tesseract.
added launch file
improved test_trajopt node
edited package.xml

Commit:
TrajOpt Planner (Empty Template) (#1478)

* Created EmptyPlan, a MoveIt planning plugin template

* added trajopt from example_planner

* fixed trajopt_planner_mannager

* modified emaptyplan_planner_manager.cpp

* applied Dave's modifications

* The rest of the modifications from Dave's comments

* removed moveit_ros_planning dependency from package.xml

* fixed clangformat error and dependency issu

* added include directory

* edited package.xml

* added include

* remove include

* remove version_gte from package.xml

* updated the setup assitant to work with trajopt planner

* removed unsuded part in initialize() from trajopt_planner_manager.cpp, kept the initialize() itself for later use

* Update moveit_planners/trajopt/src/trajopt_planner_manager.cpp

Co-Authored-By: Dave Coleman <dave@picknik.ai>

* added README
added description to trajopt_interface_plugin_description.xml

Commit:
updated resp req
updated planning context
working trajopt without any velocity constraint

ommmid committed:
added trajopt_interface class

Commit:
changed the architectrue and used TrajoptInterface class to contain the solve function details. Created a TrajOptInterface object in TrajoptPlanningContext class instead

Commit:
added kinematic_terms
in problem_description, added comments anout term_type
added ProblemInfo, TermInfo, InitInfo, BasicInfo
in kinematic_terms, added  operator()
added ConstructProblem
added some param for problem_info

Commit:
moved rotVec and concat from kinematics_term header to implementation
in problem_Description, added term_type
used MotionPlaneRequest from planning_interface instead that of moveit_msgs for solve function input
added generateInitiaTrajectory
added basic_info to problem_info

Commit:
added panda_omple_planning
added checkParameterSize function in problem_description
added JointPoseTermInfo::hatch in problem_description
added res.error.code.val to check the request in trajopt_interface.cpp
convert req.goal_constraint to JointPoseTermInfo in trajopt_interface.cpp
feed the response with the trajopt solution in trajopt_interface.cpp
remove comments from trajopt_planning_context.cpp

Commit:
addressed the comments
fixed start state
corrected the yaml file
multiple joint constraints

BryceStevenWilley commented on moveit_planners/trajopt/trajopt_interface_plugin_description.xml:
```diff
@@ -0,0 +1,7 @@
+<library path="libmoveit_trajopt_planner_plugin">
+  <class name="trajopt_interface/TrajOptPlanner" type="trajopt_interface::TrajOptPlannerManager" base_class_type="planning_interface::PlannerManager">
+    <description>
+      The trajopt motion planner plugin which implements sequential convext optimization to solve motion planning problem.
```
```suggestion
      The trajopt motion planner plugin which implements sequential convex optimization to solve motion planning problem.
```

BryceStevenWilley commented on moveit_planners/trajopt/launch/gdb_settings.gdb:
```diff
@@ -0,0 +1,8 @@
+set breakpoint pending on
+
+## Example break point:
+# break CPP_EXECUTABLE_NAME.cpp:52
```
Don't include files like this in the repository. Should probably add it to the `.gitignore`.

BryceStevenWilley commented on moveit_planners/trajopt/package.xml:
```diff
@@ -0,0 +1,40 @@
+
+  <buildtool_depend>catkin</buildtool_depend>
+
+    <depend>pluginlib</depend>
```
Nit: no indentation here

BryceStevenWilley commented on moveit_planners/trajopt/src/kinematic_terms.cpp:
```diff
@@ -0,0 +1,55 @@
+  robot_state->setJointGroupPositions("panda_arm", dof_vals);
+  robot_state->update();
+  std::string last_link_name = robot_model->getLinkModelNames().back();
+  iteration_new_pose = robot_state->getGlobalLinkTransform(last_link_name); // the pose that gets updated in optimize() based on each new solution
```
Nit: you save `link_` in the constructor, but you don't use it here and just look at the last link. Could cause confusion for your development.

ommmid replied:
Right, I will correct this when I am working on adding Cartesian constraint.

BryceStevenWilley commented on moveit_planners/trajopt/include/kinematic_terms.h:
```diff
@@ -0,0 +1,80 @@
+  CartPoseErrCalculator(const Eigen::Isometry3d& pose,
+                        const planning_scene::PlanningSceneConstPtr planning_scene,
+                        std::string link,
+                        Eigen::Isometry3d tcp = Eigen::Isometry3d::Identity())
```
tcp is usused, should be added to the `target_pose_inv * iteration_new_pose` expression below.

ommmid replied:
I am not sure about using tcp as I am using getGlobalLinkTransform to get the global transform of the end-effector. I will investigate this more when I try to add Cartesian pose constraint

BryceStevenWilley commented on moveit_planners/trajopt/include/kinematic_terms.h:
```diff
@@ -0,0 +1,80 @@
+vector<T> concat(const vector<T>& a, const vector<T>& b)
+{
+  vector<T> out;
+  vector<int> x;
```
small optimization: `vector<int> x(a.size() + b.size());`, as you do above.

BryceStevenWilley commented on moveit_planners/trajopt/include/problem_description.h:
```diff
@@ -0,0 +1,284 @@
+*/
+
+
+struct TermInfo;
```
Can use MoveIt Macros here: https://github.com/ros-planning/moveit/blob/master/moveit_core/macros/include/moveit/macros/class_forward.h

davetcoleman replied:
you don't need both line 42 and 44, the macro declares it for you

ommmid replied:
done

BryceStevenWilley commented on moveit_planners/trajopt/include/problem_description.h:
```diff
@@ -0,0 +1,284 @@
+  {                                                                                                                    \
+    TermInfoPtr out(new classname());                                                                                  \
+    return out;                                                                                                        \
+  }
```
Not a fan of this Macro, just use `TermInfoPtr out(new classname());` inline.

davetcoleman replied:
Agreed, macros should be avoided:

> Avoid defining macros, especially in headers; prefer inline functions, enums, and const variables. Name macros with a project-specific prefix. Do not use macros to define pieces of a C++ API.

https://google.github.io/styleguide/cppguide.html

ommmid replied:
done

BryceStevenWilley commented on moveit_planners/trajopt/include/trajopt_interface.h:
```diff
@@ -0,0 +1,103 @@
+/*********************************************************************
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2012, Willow Garage, Inc.
```
Believe this should be Picknik copyright.

BryceStevenWilley commented on moveit_planners/trajopt/src/trajopt_interface.cpp:
```diff
@@ -0,0 +1,402 @@
+    problem_info.basic_info.dt_lower_lim = 100.0;
+    problem_info.basic_info.start_fixed = false; // with false, we do not get any error;
+    problem_info.basic_info.use_time = false;
+    problem_info.init_info.type = trajopt_interface::InitInfo::STATIONARY;
```
`JOINT_INTERPOLATED` will be a better default here, I don't think `STATIONARY` will solve many problems.

BryceStevenWilley commented on moveit_planners/trajopt/src/trajopt_interface.cpp:
```diff
@@ -0,0 +1,402 @@
+    problem_info.basic_info.n_steps = steps_per_phase * 2;
+    problem_info.basic_info.dt_upper_lim = 2.0;
+    problem_info.basic_info.dt_lower_lim = 100.0;
+    problem_info.basic_info.start_fixed = false; // with false, we do not get any error;
```
Should be fixed. Otherwise, the starting point of the trajectory will not be the current position of the robot, and the trajectory won't be useful anymore.

ommmid replied:
Sure, it makes sense. I will work on it.

Levi-Armstrong replied:
This depends on the purpose. For example, if you are trying to execute a process like sanding the critical part is the actual process path not how you get to the start of the process path. So you plan the process path first leaving the start free to hopefully get the most optimal and then you plan from the current location with start fixed to the start of the process path. It depends on what you want the default behavior to be.

BryceStevenWilley commented on moveit_planners/trajopt/src/trajopt_interface.cpp:
```diff
@@ -0,0 +1,402 @@
+    int steps_per_phase = 10;
+    problem_info.basic_info.n_steps = steps_per_phase * 2;
+    problem_info.basic_info.dt_upper_lim = 2.0;
+    problem_info.basic_info.dt_lower_lim = 100.0;
```
Is there a reason your lower limit is higher than the upper limit? I see that `use_time` is false, but still seems like a weird default.

ommmid replied:
1/dt is used for optimization.

davetcoleman replied:
Please add this note as a comment in line for future readers who have the same question

BryceStevenWilley commented on moveit_planners/trajopt/src/trajopt_interface.cpp:
```diff
@@ -0,0 +1,402 @@
+    std::cout << " ======================================= create ProblemInfo"  << std::endl;
+    ProblemInfo problem_info(planning_scene, req.group_name);
+
+    int steps_per_phase = 10;
```
Not an issue now, but this param will need to be configurable, either by some internal algorithm or by the user.

davetcoleman replied:
@ommmid please take feedback like this and note as TODO in the code

BryceStevenWilley commented on moveit_planners/trajopt/src/trajopt_interface.cpp:
```diff
@@ -0,0 +1,402 @@
+      // Create place pose constraint
+      std::shared_ptr<trajopt_interface::JointPosTermInfo> joint_pos(new trajopt_interface::JointPosTermInfo);
+      joint_pos->term_type = trajopt::TT_CNT;
+      joint_pos->name = "joint pose name ^_^";
```
:laughing:

BryceStevenWilley commented on moveit_planners/trajopt/src/trajopt_interface.cpp:
```diff
@@ -0,0 +1,402 @@
+
+
+    // assume that the trajectory is now optimized, fill in the output structure:
+    ROS_DEBUG_NAMED("chomp_planner", "Output trajectory has %d joints", opt_solution.rows());
```
```suggestion
    ROS_DEBUG_NAMED("trajopt_planner", "Output trajectory has %d joints", opt_solution.rows());
```

BryceStevenWilley commented on moveit_planners/trajopt/src/trajopt_interface.cpp:
```diff
@@ -0,0 +1,402 @@
+    moveit::core::RobotState start_state(planning_scene->getCurrentState());
+    moveit::core::robotStateMsgToRobotState(req.start_state, start_state);
+    // This function converts the moveit_msgs::RobotState (from req.start_state) to moveit::core::RobotState (start_state).
+    start_state.update();
```
Your PR comment mentions that the start state is not set. How is anything working if you don't set the start state? If you are only setting the goals state, then none of the trajectories you get should be feasible.

ommmid replied:
I meant the start state as a constraint for trajopt as we talked in the previous meeting.

BryceStevenWilley replied:
Ah, I see. I think I misunderstood what you meant by 'somewhat working': I assumed that you had gotten valid trajectories on some examples and had fixed the issue we talked about on Thursday.

ommmid replied:
I added the the start state as a constraint too.

BryceStevenWilley reviewed (commented):
Just some initial thoughts, most comments you can ignore until this is ready for a full review, but most of the comments on the TrajOpt interface document are important.

davetcoleman commented:
> start_fixed is set to false. if it is true it means: it constraints the joint values to be their initialized ones but because the start state is not set, then it creates problem. I am not sure about this.

This seems like a really important detail to understand. Have you been able to read through its behavior in Tesseract to understand what it does? Maybe @Levi-Armstrong could quickly shed some light on this variable

davetcoleman commented on moveit_planners/trajopt/config/setup.yaml:
```diff
@@ -0,0 +1,14 @@
+trajopt_param:
```
please add comments explaining what each of these parameters do, so that reviewers can better understand this PR

davetcoleman replied:
You may want to cite the publication / note the paper's notation in the comments

ommmid replied:
yaml file is the right place to put the citation or it should be in another file?

davetcoleman replied:
sorry, i didn't mean cite the paper, i meant explain how these parameters relate to what were presented in the original research, since the var names probably differ

davetcoleman commented on moveit_planners/trajopt/include/kinematic_terms.h:
```diff
@@ -0,0 +1,80 @@
+ * @brief Used to calculate the error for StaticCartPoseTermInfo
+ * This is converted to a cost or constraint using TrajOptCostFromErrFunc or TrajOptConstraintFromErrFunc
+ */
+struct CartPoseErrCalculator : public sco::VectorOfVector
```
Have you used rviz_visual_tools to better understand if your vectors are in the right frame, and if the error calculation intuitively makes sense?

davetcoleman replied:
I've never seen a struct inheritance... this isn't normal

ommmid replied:
I got it from the original trajopt pakage. Should I change them to class?

ommmid replied:
I am actually using rviz_visual_tools

davetcoleman replied:
yes, per Google style, if it has functions, it should be a class

davetcoleman commented on moveit_planners/trajopt/include/problem_description.h:
```diff
@@ -0,0 +1,286 @@
+MOVEIT_CLASS_FORWARD(TermInfo);
+
+class TrajOptProblem;
+typedef std::shared_ptr<TrajOptProblem> TrajOptProblemPtr;
```
use the moveit macros here, too. they combine lines 46 and 47

davetcoleman commented on moveit_planners/trajopt/include/problem_description.h:
```diff
@@ -0,0 +1,286 @@
+
+enum TermType
+{
+  TT_COST = 0x1,      // 0000 0001
```
please add comments explaining what these are

ommmid replied:
They are explained in line 15 and below that

davetcoleman commented on moveit_planners/trajopt/include/problem_description.h:
```diff
@@ -0,0 +1,286 @@
+
+struct BasicInfo
+{
+  /** @brief If true first time step is fixed with a joint level constraint*/
```
I have a hard time understanding the usefulness of a motion plan without a fixed start state. Why is this an option?

ommmid replied:
https://github.com/ommmid/moveit/pull/1/files/80aafb69eba5b5d31a769c37fcb26c45326f568f#r306575008

davetcoleman commented on moveit_planners/trajopt/include/problem_description.h:
```diff
@@ -0,0 +1,286 @@
+  bool start_fixed;
+  /** @brief Number of time steps (rows) in the optimization matrix */
+  int n_steps;
+  std::string manip;
```
use better variable names... what is this?

ommmid replied:
I think it is equivalent to robot_model. It is not used because I use planning scene. I remove it.

davetcoleman commented on moveit_planners/trajopt/include/problem_description.h:
```diff
@@ -0,0 +1,286 @@
+  /** @brief Number of time steps (rows) in the optimization matrix */
+  int n_steps;
+  std::string manip;
+  std::string robot;             // optional
```
better var name

ommmid replied:
not used, removed

davetcoleman commented on moveit_planners/trajopt/include/problem_description.h:
```diff
@@ -0,0 +1,286 @@
+};
+
+
+struct ProblemInfo
```
Have you tried constructing a very very basic problem, manually, to ensure you understand how this works?

Like a 1dof arm or something?

ommmid replied:
It is actually a good idea. It will help me understand the algorithm more in details. I will do it soon.

davetcoleman commented on moveit_planners/trajopt/include/problem_description.h:
```diff
@@ -0,0 +1,286 @@
+  TrajOptProblem();
+  TrajOptProblem(const ProblemInfo& problem_info);
+  virtual ~TrajOptProblem() = default;
+  sco::VarVector GetVarRow(int i, int start_col, int num_col) { return m_traj_vars.rblock(i, start_col, num_col); }
```
Add comments for these functions

davetcoleman commented on moveit_planners/trajopt/include/problem_description.h:
```diff
@@ -0,0 +1,286 @@
+
+  Set term_type == TT_COST or TT_CNT for cost or constraint.
+*/
+struct CartPoseTermInfo : public TermInfo
```
Do you need this struct current? Since you arent' supporting cartesian planning yet?

ommmid replied:
Actually the cartesian constrain is implemented too but I am not sure about its validity yet. I make sure it is working correctly, otherwise will remove it and make another PR for that.

davetcoleman commented on moveit_planners/trajopt/include/trajopt_interface.h:
```diff
@@ -0,0 +1,103 @@
+/*********************************************************************
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2012, PickNik, Inc.
```
change year

davetcoleman commented on moveit_planners/trajopt/include/trajopt_interface.h:
```diff
@@ -0,0 +1,103 @@
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of Willow Garage nor the names of its
```
change company

davetcoleman commented on moveit_planners/trajopt/src/kinematic_terms.cpp:
```diff
@@ -0,0 +1,55 @@
+  Quaterniond q(pose_err.rotation());
+  VectorXd err = concat(Vector3d(q.x(), q.y(), q.z()), pose_err.translation());
+  return err;
+}
```
unit testing for functionality like this would greatly reduce potential issues. are there already unit tests for it in the upstream project that you can bring in and adapt?

ommmid replied:
not that I know of.

ommmid replied:
I will try to make one

davetcoleman commented on moveit_planners/trajopt/src/test_trajopt.cpp:
```diff
@@ -0,0 +1,237 @@
+#include <moveit/robot_model/robot_model.h>
+#include <moveit/robot_state/robot_state.h>
+
+// test
```
add better description of what this file does

davetcoleman commented on moveit_planners/trajopt/src/test_trajopt.cpp:
```diff
@@ -0,0 +1,237 @@
+int main(int argc, char** argv)
+{
+    const std::string node_name = "trajopot_planning_tutorial";
+      ros::init(argc, argv, node_name);
```
always run clang-format before making a PR

davetcoleman commented on moveit_setup_assistant/src/widgets/configuration_files_widget.cpp:
```diff
@@ -375,6 +375,18 @@ bool ConfigurationFilesWidget::loadGenFiles()
   file.write_on_changes = 0;
   gen_files_.push_back(file);
 
+  // trajopt_planning_pipeline.launch
```
Let's keep changes to the MSA out of this PR, please put them in a separate PR

ommmid replied:
Yes they should be in another PR.

davetcoleman reviewed (commented):
Can you share relevant diagrams explaining the relationship of the files / classes in this PR?

ommmid commented:
> > start_fixed is set to false. if it is true it means: it constraints the joint values to be their initialized ones but because the start state is not set, then it creates problem. I am not sure about this.
> 
> This seems like a really important detail to understand. Have you been able to read through its behavior in Tesseract to understand what it does? Maybe @Levi-Armstrong could quickly shed some light on this variable

https://github.com/ommmid/moveit/pull/1#discussion_r306332030

ommmid commented:
> Can you share relevant diagrams explaining the relationship of the files / classes in this PR?

![trajoptflow](https://user-images.githubusercontent.com/26231820/61745206-36d17480-ad56-11e9-9353-9a609bace408.png)

Commit:
addressed some comments
leave setup assistant changes for later on another PR
cleaned up test_trajopt
added unittest

Commit:
added kinematic DOF of the active joint model group to TrajOptProblem class
added name_ for ROS_DEBUG_STREAM_NAMED
addressed the comments
moved launch file to moveit_tutorial
moved yaml file to panda_moveit_config
used bullet for collision checking

Commit:
Added comments about initial trajectory
changed my email address
renamed rotVec and concat
added tesseract license

Commit:
clang format

Commit:
applied Henning's comments

Commit:
added catkin ignore to ignore this package for now as it has dependencies. The next PR wont have any dependencies

Commit:
addressed the comments

Commit:
renamed hatch to addObjectiveTerms

Commit:
addressed Dave's comments

Commit:
added catkin_ignore

Commit:
prepare for merge

Commit:
resolved conflicts

Commit:
clang format

Commit:
removed trajopt_test_support from package.xml

Commit:
fixed travis errors

Commit:
addressed Dave's comments

JafarAbdi committed:
Enable installing .hpp files

JafarAbdi committed:
Fix catkin_lint errors

JafarAbdi committed:
Update included header files - Correct namespace

JafarAbdi committed:
Update CMakeLists.txt

JafarAbdi committed:
Update package.xml

Commit:
added README.md to trajopt package

Commit:
marked override

Commit:
added the reference trajectory; before, it was hard-coded

Commit:
find osqp locally

mamoll committed:
remove unlicensed binaries for BPMPD, make OSQP the default QP solver

Since OSQP is not available via apt and the dependency on OSQP is not
exposed, we just download and install OSQP in the build dir via CMake's
ExternalProject module.

Commit:
fixed license

Commit:
clang format

Commit:
correctedd num_steps in the loop

Commit:
When using motion planning display in rviz, the reference trajectory is not set. Check this and warn the user if ref traj is empty
trajectory_test not working yet

Commit:
run demo from panda_moveit_config befire running the test

Commit:
Fixed the test. changed the type to interpolated.

Commit:
clang format

Commit:
moved all the files from pand_mvoit_config and franka_description into the test

Commit:
Update moveit_planners/trajopt/trajopt/CMakeLists.txt

Co-Authored-By: Ayush Garg <ayushgargdroid@gmail.com>

Commit:
Update moveit_planners/trajopt/trajopt/CMakeLists.txt

Co-Authored-By: Ayush Garg <ayushgargdroid@gmail.com>

Commit:
Update moveit_planners/trajopt/trajopt/package.xml

Co-Authored-By: Ayush Garg <ayushgargdroid@gmail.com>

Commit:
rebased

Commit:
osqp installed from trajopt_utils instead of trajopt_sco

osqp_FOUND is false if osqp is pulled in trajopt_sco. osqp can not be found from the same CMakeLists that was used to pull down its repo and install it in build directory. A workaround is to install osqp from trajopt_utils and check if it exists in trajopt_sco. trajopt_sco is dependent on trajopt_utils

Another fix:  trajopt_problem_ from trajopt_interface needed to be corrected as a shared_pointer.

Commit:
download osqp during configuration - from Jafar's PR

jrgnicho committed:
rviz panel: widgets for adding geometric primitives

Commit:
working on CollisionCost

Commit:
working on finding the right value for distances between collision objects. Contact type has a depth memeber but in tesseract, they use distance member in their ContactResult type

Commit:
adjusted more parts of the code for MoveIt but what is left is how I calculate cc_nearest_points in MoveIt, Contact type does not have such a thing

rhaschke committed:
Prefer vendor-specific OpenGL library

Co-authored-by: Henning Kayser <henningkayser@picknik.ai>

rhaschke committed:
use SYSTEM includes to suppress external warnings

rhaschke committed:
Noetic docker files

rhaschke committed:
Fix warnings

rhaschke committed:
reorder and group clang-format options

rhaschke committed:
add new config options

rhaschke committed:
Tweak penalties

Allow some excess characters per line if that improves readability.

rhaschke committed:
Travis: Enable Noetic build

rhaschke committed:
Remove moveit_noetic.rosinstall

rhaschke committed:
Further increase acceptance threshold for mesh-filter test

Mesa OpenGL 20.x seems to have worsen the situation...

v4hn committed:
partially transition unused test bin to rostest (#2158)

There is a lot of infrastructure missing to get this to actually run...
- the plugin needs to be registered
- pr2_moveit_config is not in moveit_resources

felixvd committed:
Clean up Rviz plugin, fix mesh scaling (#2142)

* Clarify object importing

* Add suggestion (LARGE_MESH_THRESHOLD)

* Add file filter

* Add tooltip, clarify names

* Fix and clarify names in robot states tab

* Add dialogue to Clear button in robot states

pbeeson committed:
Parameterize input trajectory density of Time Optimal trajectory generation  (#2185)

MoveIt wrapper to Time Optimal Trajectory Generation downsamples trajectory, but its hard coded to keep waypoints with at least 0.001 radians of motion on a joint.  For very long and/or dense input trajectories, the downsampled trajectory is still way too large for the iterative optimization algorithm to finish in reasonable time.  I was seeing 3000 point trajectory taking ~15 seconds.  This fix allows the user to set this downsampling density in minimum radians of motion between two waypoints to something else (presumably higher like 0.5 degrees [0.00875 radians]) in order to drastically improve the speed of this algorithm.

AndyZe committed:
Fix a parameter mix-up in moveit_cpp loading (#2187)

AndyZe committed:
moveit_servo: delete python integration tests (#2186)

which (mostly) are redundant to the corresponding C++ tests.

henningkayser committed:
Fix segfault in PSM::clearOctomap() (#2193)

tylerjw committed:
use moveit_resources for testing (#2194)

rhaschke committed:
Unify UI of adding primitive and mesh shapes

- Remove "Import from File" and "Import from URL" buttons,
  but add those options as item in the shape combobox
- Integrate MotionPlanningFrame::addObject() into MotionPlanningFrame::addSceneObject()

rhaschke committed:
MP plugin: Tweak GUI layout

- grid layout for object position/orientation spin boxes
  (to not rely on chance for their column-wise layout anymore)
- Limit position spin boxes to -999..999
  This essentially cancels one digit to be displayed, which significantly reduces the
  minimum width of the widget.

rhaschke committed:
Update query states if a collision object was attached/detached

rhaschke committed:
Rename MotionPlanningFrame functions

clearSceneButtonClicked() -> clearScene()
removeObjectButtonClicked() -> removeSceneObject()
publishSceneButtonClicked()-> publishScene()

rhaschke committed:
Track local PS changes and publish them before planning

v4hn committed:
Remove moveit_resources from CMakeLists.txt

There is no reason to include them from cmake anymore.
The resources are not in a single path anymore and there is no C header exposed anymore.

rhaschke committed:
Replace moveit_resources/<folder> with moveit_resources_<package>

rhaschke committed:
Default to Planning tab initially

rhaschke committed:
Fix loading of test resources

- Convert code to use robot_model_test_utils
- Fixup loading in robot_model_test_utils

rhaschke committed:
Adapt expected self-collision distance

rhaschke committed:
Explicitly added icon resources

JafarAbdi committed:
Remove urdf package as build_depend from package.xml (#2207)

rhaschke committed:
Merge PR #2180: Various fixes for upcoming Noetic release

rhaschke committed:
Merge PR #2198: Allow adding planning scene shapes from rviz panel

Add primitive shapes widgets

JeroenDM committed:
Fix docstrings in robot_state.h (#2215)

rhaschke committed:
Travis: Update dist to Bionic (#2221)

... to fix issue https://github.com/moby/moby/pull/36417

JeroenDM committed:
Add named logging with LOGNAME to OMPL interface (#2211)

* add named logging with LOGNAME to ompl interface

* clang-format

JeroenDM committed:
Remove dead code from ompl interface (related to subspaces) (#2214)

* remove unused subspaces option and code from ompl interface

* remove unused code from StateValidityChecker in ompl interface

gerardcanal committed:
Exposed missing parameters from the MoveGroup Python interface (#2201)

* Exposed the wait_for_servers parameter in the Pyhton interface

* Add missing getPlannerId interface from MoveGroup PythonWrapper

JeroenDM committed:
Add documentation to ompl interface (#2226)

JeroenDM committed:
OMPL interface: fix / ignore clang-tidy warnings (#2227)

* fix / ignore clang-tidy warnings in ompl interface

* remove NOLINT and fix warning in ompl interface

mvieth committed:
Optimize includes (#2229)

Remove most obvious unused includes, as recommended by include-what-you-use (see https://github.com/ros-planning/moveit/issues/2170)

AndyZe committed:
Remove unused yaml param (#2232)

AndyZe committed:
Check collisions during joint motions, too (#2204)

* Check collisions during joint motions, too.

Print colliding objects, for debugging.

* Small code simplification

* Use the argument to printCollisionPairs

* Clang format

* Fix misspelling

* Log all colliding pairs

* Adjustments to print formatting and log levels

felixvd committed:
Fix subframes disappearing when object is detached/scaled/renamed (#1866)

* Treat subframes when detaching objects

* reduce code complexity

- remove code duplication, when fetching attached bodies
- simplify function calls when detaching (attached) bodies

* Maintain subframes when scaling/renaming objects in Rviz

* clang-format

* Add review comments

* More review comments

* minor nits

* fixup: use STREAM logging

* Add TODO

Co-authored-by: Robert Haschke <rhaschke@techfak.uni-bielefeld.de>
Co-authored-by: v4hn <me@v4hn.de>

v4hn committed:
RViz RS & Traj display: ensure robot_description is available onEnable

This is a (in my opinion) rather dirty workaround because RViz
enables Displays systematically *before* loading their properties.

So the patch loads the required property a second time *before* everything else.

Assuming this ever changes we can remove this code again.

A Display's "enabled" state is encoded in a boolean property the Display inherits from.
The way loading works (recursively running through the properties)
this special property is loaded first and triggers onEnable because
of the Property's changed() signal.

A second "Enabled" property exists in RViz's Displays but does not change behavior at all.

JeroenDM committed:
Fix memcpy bug in copyJointToOMPLState in ompl interface (#2239)

tnaka committed:
Add planning_adapters for chomp in setup assistant (#2242)

CHOMPPlanner doesn't fill velocity, acc or time.
We need AddTimeParameterization at least to use this planner with
ros_control.

rhaschke committed:
Revert "delay loading of robot model to avoid spurious warning"

This reverts commit 4e72ac5f96863e7c37a47706645c6634d7672f7b
as the delayed loading results in link properties NOT being loaded.

rhaschke committed:
Only loadRobotModel() if not yet done

rhaschke committed:
Switch docker image to clang-format-10

rhaschke committed:
apply changes

rhaschke committed:
Run clang-format on Noetic image (with clang-format-10 installed)

rhaschke committed:
Migrate to clang-format-10

AndyZe committed:
A ROS service to reset the Servo status (#2246)

rhaschke committed:
Merge PR #2199: Adapt repository for splitted moveit_resources layout

rhaschke committed:
Temporarily add moveit_resources to moveit.rosinstall

... as long as moveit_resources is not yet released

rfn123 committed:
Correctly set velocities to zero when stale (#2255)

v4hn committed:
Merge pull request #2244 from ubi-agni/fix-robot-model-loading

Fix deferred robot model loading

v4hn committed:
MP display: add units to joints tab (#2264)

ymollard committed:
Start new joint_state_publisher_gui on param use_gui (#2257)

tylerjw committed:
add soname version to moveit_servo (#2266)

davetcoleman committed:
Move constraint representation dox to moveit_tutorials (#2147)

gerardcanal committed:
Add missing variants of place from list of PlaceLocations and Poses in the python interface (#2231)

gerardcanal committed:
Added GILRelease to pick and place (#2272)

AdamPettinger committed:
Update last_sent_command_ at ServoCalcs start (#2249)

* Move last_sent_command_ update from constructor to start
* Tweak demo configs
* Add install folders to CMakeLists
* Add clarity to parameter warning

AndyZe committed:
Add a utility to print collision pairs (#2275)

* Add a utility for logging collision pairs
* Update printing style, rename function, move constants into the struct

JeroenDM committed:
fix some spelling errors in the ompl interface (#2279)

rhaschke committed:
Cleanup: Add a utility to print collision pairs (#2278)

rhaschke committed:
Replace $(find xacro)/xacro -> xacro (#2282)

tylerjw committed:
adjust code coverage yaml to make it more forgiving (#2283)

tylerjw committed:
[maint] Remove kinetic from Travis CI testing (#2295)

rhaschke committed:
Temporarily add srdfdom to moveit.rosinstall

... until it is released.

rhaschke committed:
Revert "Temporarily add srdfdom to moveit.rosinstall"

This reverts commit 4db7ccb795ca8efa8de9646efa01b72bf1ff62ad.
Instead, use ros-testing repo.

tylerjw committed:
1.1.0

v4hn committed:
MSA: only write default_planner_config field if any is selected (#2293)

Otherwise, we get entries like this in ompl_planning.yaml, where "None", of course, is not a valid planner config:

arm:
    default_planner_config: None

rhaschke committed:
Replace panda_moveit_config -> moveit_resources_panda_moveit_config (#2300)

This fixes a failing test on the ROS buildfarm: http://build.ros.org/job/Ndev__moveit__ubuntu_focal_amd64/1

tylerjw committed:
[maint] Remove patch status from codecov (#2306)

felixvd committed:
Update codeowners file (#2309)

tylerjw committed:
[fix] Servo heap-buffer-overflow bug (#2307)

felixvd committed:
Add comment to MOVEIT_CLASS_FORWARD (#2315)

ymollard committed:
robot_description is already loaded in move_group.launch (#2313)

AndyZe committed:
Refactor velocity limit enforcement and add a unit test (#2260)

* Rework velocity limit enforcement, add unit test

* Test every command out of Servo

* Minor variable renaming

* Clang format & clang tidy

* Delete unused acceleration variable

v4hn committed:
Noetic was released (#2304)

AndyZe committed:
Fix Servo thread interruption (#2314)

v4hn committed:
MGC: Improve exception messages (#2318)

LoyVanBeek committed:
Let the max number of contacts be the amount of world objects + link models with geometry

LoyVanBeek committed:
Add tests for checking whether an empty world + self-collision list any contacts

felixvd committed:
Clean up Rviz Motion Planning plugin, add tooltips (#2310)

- better use of space / overall smaller panel

- remove goal constraints field, it's not used and not intuitive in the GUI either

- reorder check boxes by subjective usefulness

- Add better tooltips

- Remove autoScroll from Planning Parameters field

- Fix tabstop order

- Align columns of Planning tab

Co-authored-by: Robert Haschke <rhaschke@techfak.uni-bielefeld.de>

rhaschke committed:
ikfast: add missing python dependencies

rhaschke committed:
Install python scripts via catkin_install_python()

... to correctly rewrite shebang

rhaschke committed:
Remove shebang line from moveitjoy_module.py

v4hn committed:
Merge pull request #2323 from ubi-agni/fix-ikfast

fix some more python3 issues

rhaschke committed:
Adapt to API changes in geometric_shapes (#2324)

https://github.com/ros-planning/geometric_shapes/pull/161 introduced a syntax simplification employing C++14 features (constexpr).

rhaschke committed:
Workaround python3 issue with lxml.ElementTree.write()

Fixes https://github.com/ros-planning/moveit/issues/2322

rhaschke committed:
Fix some more issues with create_ikfast_moveit_plugin.sh

- require python-lxml and python-yaml to be installed
- use str() to print Exception

rhaschke committed:
Travis: default to Noetic

rhaschke committed:
Remove custom copy-assignment operator for DistanceResultsData

It was just performing the operations of the default operator. Newer clang requires either both or none of copy constructor and assignment operator to be defined.

warning: definition of implicit copy constructor for 'DistanceResultsData' is deprecated because it has a user-declared copy assignment operator [-Wdeprecated-copy]

rhaschke committed:
IKFast: suppress unused-parameter warnings

rhaschke committed:
perception: add missing dependency to OpenMP

rhaschke committed:
Silent deprecation warnings of gtest's typed test suites

rhaschke committed:
Handle multiple link libraries for FCL (#2325)

gavanderhoorn committed:
Merge pull request #2327 from ubi-agni/fix-ikfask-tests

Fix various issues with Noetic build

rhaschke committed:
Fix clang-tidy warning (#2334)

rhaschke committed:
MSA: Fix disappearing robot on change of reference frame (#2335)

- set visibility of robot-state-display to true after reset()
- correctly update reference frame also on deletion of virtual joint

rhaschke committed:
Fix some clang-tidy issues on Travis (#2337)

* Replace typedefs with using declarations
* Move default destructor definitions to headers
* Silent spurious clang-tidy warning
* Move variable definitions to their usage location

tylerjw committed:
[fix] Segfault when editing pose in moveit_setup_assistant (#2340)

tylerjw committed:
Changelogs for Noetic 1.1.1

felixvd committed:
Fix "Clear Octomap" button, disable when no octomap is published (#2320)

* Enable/disable "Clear Octomap" button
* Correctly remove octomap from scene in clearOctomap()

AndyZe committed:
moveit_cpp: more informative error message, cover another potential failure condition. (#2336)

rhaschke committed:
MSA: Update RobotState *before* visualization

rhaschke committed:
remove commented, copy-pasted code

rhaschke committed:
Use PlanningScene's RobotState for postures

.. instead of extra joint_state_map_

rhaschke committed:
Reorder buttons in GroupEditWidgets

PeterMitrano committed:
Fix ROS namespacing in moveit_commander's PSI (#2347)

rhaschke committed:
Replace dummy QWidgets with QSpacerItems

v4hn committed:
Merge pull request #2348 from ubi-agni/msa-fix-poses

MSA: fix pose definitions

rhaschke committed:
Use QStackedWidget instead of QWidget+QStackedLayout

rhaschke committed:
Increase minimal window size

rhaschke committed:
Cleanup Qt includes

rhaschke committed:
Cleanup ROS includes

rhaschke committed:
Remove diagnostics macros

This is more elegantly handled by system includes now.

rhaschke committed:
Fix unused-parameter warnings

rhaschke committed:
Remove released upstream packages from docker files

As franka_description is released, we don't need to build it in our docker images anymore.

rhaschke committed:
Merge PR #2349: Fix MSA layout

rhaschke committed:
MSA: Allow showing both, visual and collision geometry (#2352)

rhaschke committed:
Merge PR #2351: [maint] Cleanup MSA includes

rhaschke committed:
MSA: Fix group editing (#2350)

Group Editing: Cancel falls back to previous screen

When entering joints/links/chain/subgroup edit screen from "Add Group" screen,
cancel was cancelling the whole operation, loosing all info that was already entered for the group.
Now it just returns to the previous screen and thus allows to continue editing the new group.

v4hn committed:
move_group: simplify test launch files

v4hn committed:
Merge pull request #2355 from LoyVanBeek/fix_master/contacts_in_empty_world

[Master] Let the max number of contacts be the amount of world objects + link models with geometry

Commit:
A library for servoing toward a moving pose (#2203)

* A library for servoing toward a moving pose

* Use the control_toolbox version of PID control

* Minor cleanup after switching to control_toolbox PID

* Use default ROS1 library type

* Decrease the default windup limit

* Add a basic unit test

* Add orientation tracking, too

* Warning if no target pose is provided

* One PID for orientation instead of 3

* A different method of ensuring a new target pose is received

* Multithread the test pose publisher

* Correct service namespaces

* Clang format and clang tidy

* Minor tweaks for CI

* Add function to update PID settings

* Add a method to get PID errors

* Be consistent with service namespacing

This is important if there are multiple arms Servoing simultaneously

* Address Adam's more minor comments

* First attempt at using pose tracking

* Small updates to launch files and CMakeLists

* Address Andy comments

* Get Pose Tracking demo working (#3)

* Rework command frame name and target pose

* Add EE frame tf and make it available thru C++ interface

* Pass getEEFrameTransform thru PoseTracking

* Update TF calcs at servo start

* Modify pose demo to move slightly from current position

* Make PoseTracking use EE tf instead of command frame tf

* Clean up the demo

* Make helper func for converting eigen to tf message

* Make the Servo instance public, to access functions like setPaused()

* Use an enum class for status codes, so entries can be duplicated

* Add a timeout argument

* Clang format

* Fix a race condition with initial joint reception

* Update config file for testing

* Wrap tf lookup in a try/catch block

* Use a tf2 utility for Eigen->msg conversion

* Explicitly initialize transforms to zero

* More robust test setup

* Update the command frame at runtime

* Update angular PID configs, too

* Properly wait for test setup

Co-authored-by: AdamPettinger <adam.pettinger@utexas.edu>

rhaschke committed:
Depend on ros-noetic-fcl (0.6) in Noetic (#2359)

* Depend on ros-noetic-fcl (0.6) in Noetic
* Travis: switch to ros-testing repos

gleichdick committed:
Make GILReleaser exception-safe (#2363)

If an exception occurs while the GIL is released, it won't be reacquired and the stack unwinding will take place
without the GIL. GILReleaser::release() and GILReleaser::reacquire() is removed, because it is unsafe.

tylerjw committed:
1.1.1

gleichdick committed:
Unit Test for ByteString-based ROS msg conversion (#2362)

PeterMitrano committed:
Python interface improvements. Fix #1966, add enforceBounds (#2356)

* implement get state and get state bounded

* add enforceBounds

Co-authored-by: Peter <pmitrano@storm20.04>

nbbrooks committed:
Fix ordering of windup args to control_toolbox::Pid (#2370)

DLu committed:
MSA launch files: fix indentation (#2371)

converting tabs to spaces

JeroenDM committed:
Add test for PlanningContextManager in ompl interface (#2248)

DLu committed:
Update .github files (#2380)

Commit:
Fix servo trajectory point timestamping (#2375)

Commit:
Change servo namespacing logic (#2354)

* Publish services correctly in the namespace of the node.

* Set relative joint state topic names be relative to the parent namespace of the servo node.

* Make parameter sub-namespace optional. By default load the parameters into node namespace.

* Fix node handles for the servo server node and the cpp interface example to use node namespace instead of the parent namespace.

* Fix formatting

* Reading ee_frame_name parameter got lost somewhere along the way. Add it back.

* Remove explicit parameter_ns variables and used namespaced node handles instead.

* Fix internal namespace to truly be internal even in case where two or more Servo instances are launched from a single node.

* Fix command_out_topic parameter in ur_simulated_config.yaml example config.

* Apply namespacing fixes also to tests.

* Shorten the comment about joint states topic names.

tylerjw committed:
Cleanup current state handling in servo (#2372)

JeroenDM committed:
Add test to ompl interface for StateValidityChecker (#2247)

* add generic robot test class to ompl interface test folder

* add test for ompl interface StateValidityChecker

* add test with path constraints to ompl interface StateValidityChecker

* use debug log statements, add newline, cleanup comments

* remove dead code and update comment

* Change company name in license

* update state validity test comment

* improve assert statements

* remove empty line from load_test_robot.h

auman66 committed:
Fix broken documentation link in README (#2393)

Commit:
Fix pose tracking race condition (#2395)

* Stop rewriting target_pose_ timestamps. It is caller's responsibility to ensure timestamp is correct and recent enough.

* Prevent data races on target_pose_ by limiting access to it with a mutex.

* Re-arrange part in targetPoseCallback function to make the callback lightweight when the target pose is in planning frame.

* Small improvements to comments and code readability.

* Set mutex as mutable.

* Fix formatting.

* Improve comment re. angle-axis PID tracking

* Implement resetTargetPose() and add it to test & example

* Add mutex protection to resetTargetPose()

* Add missing semicolon and fix formatting.

Co-authored-by: AndyZe <andyz@utexas.edu>

ct2034 committed:
Add Pilz industrial motion planner (#1893)

* Add pilz_industrial_motion_planner to moveit_planners 
* Update license, document consent to this change by contributers
* Fix formatting, code style, catkin_lint, clang-tidy
* Add and update tests, use new prvt_moveit_config from moveit_resources
* New codeowners for pilz_industrial_motion planner
* Add pipeline configuration teimplates to MSA

Co-authored-by: Joachim Schleicher <J.Schleicher@pilz.de>
Co-authored-by: rfeistenauer <r.feistenauer@pilz.de>
Co-authored-by: Giuseppe Sansone <g.sansone@pilz.de>
Co-authored-by: Immanuel Martini <i.martini@pilz.de>

rhaschke committed:
FixStartStateBounds: Copy attached bodies when adapting the start state (#2398)

tylerjw committed:
[meta] first-timers-only issue template (#2385)

Co-authored-by: Felix von Drigalski <FvDrigalski@gmail.com>
Co-authored-by: Sebastian Jahr <jahr.sebastian@t-online.de>

rhaschke committed:
Fix QObject::connect: Cannot queue arguments of type 'QVector<int>' (#2392)

AndyZe committed:
Move timer initialization down to fix potential race condition

tylerjw committed:
[maint] Changelogs in pilz planner (#2408)

Commit:
Update collision-related comments (#2382) (#2388)

Co-authored-by: Ghenohenomohe <71963599+Ghenohenomohe@users.noreply.github.com>

DLu committed:
Fix empty sequence in moveit_setup_assistant (#2406)

LegoYoda112 committed:
Changed processing_thread_ spin to use std::make_unique instead of new (#2412)

tylerjw committed:
[feature] Low latency mode (#2401)

Co-authored-by: AndyZe <andyz@utexas.edu>

davetcoleman committed:
Add Stargazer chart to README (#2417)

stuart-fb committed:
Fix OrientationConstraint::decide (#2414)

* Fix OrientationConstraint::decide

* Add new test case for OrientationConstraintsSimple

* Add comments for clarity

harshgupta246 committed:
Updated First Timers Only Issue Template (#2419)

nbbrooks committed:
Prevent moveit_servo transforms between fixed frames from causing timeout (#2418)

AndyZe committed:
Delete CollisionRequest min_cost_density

archonixX committed:
docker-source: support running clang-tidy (#2430)

JafarAbdi committed:
RobotModelBuilder: Add parameter to specify the joint rotation axis

JafarAbdi committed:
Fix validation of orientation constraints (#2434)

* Add tests that trigger the bug
* Use the angle differece between the desired and the current orientation
  to check if the constraint is satisfied
* Handle singularities

rhaschke committed:
Upgrade cmake_minimum_required to 3.1 (#2453)

to suppress CMP0048 warning

rhaschke committed:
RobotModelBuilder: Allow adding end effectors (#2454)

Commit:
Fix doxygen documentation for setToIKSolverFrame (#2461)

* Fix doxygen documentation for setToIKSolverFrame

* "Convert" -> "Transform"

* Make function private. Update comments.

* Make inline and private

* Longer function should not be inline

Co-authored-by: Robert Haschke <rhaschke@users.noreply.github.com>

Co-authored-by: Robert Haschke <rhaschke@users.noreply.github.com>

Commit:
Add an important sleep in Servo pose tracking (#2463)

DLu committed:
Missing RViz and moveit_simple_controller_manager dependencies in MSA template (#2455)

Udbhavbisarya23 committed:
Fix some typos in comments (#2466)

Commit:
Catch exceptions during RobotModel loading in rviz (#2468)

Commit:
Provide a function to set the position of active joints in a JointModelGroup (#2456)

* RobotState::setJointGroupPositions: assert correct size of  vector

fixes #2442

* setJointGroupActivePositions sets only the positions of active joints

* Add overloads

* fix asserts

* implement JointModelGroup::getActiveVariableCount

Tobias-Fischer committed:
Python3 compatibility for ikfast's round_collada_numbers.py (#2473)

AdamPettinger committed:
[Servo] Fix initial angle error is always 0 (#2464)



Co-authored-by: AndyZe <andyz@utexas.edu>
Co-authored-by: Bjar Ne <gleichdick@users.noreply.github.com>

sevangelatos committed:
Fix RobotState::dropAccelerations/dropEffort to not drop velocities (#2478)

v4hn committed:
pilz planner: add string includes (#2483)

Commit:
Clean up collision-related log statements (#2480)

ShotaAk committed:
Fix scaling factor parameter names (#2452)

bostoncleek committed:
Improve robustness of move group interface test (#2484)

rhaschke committed:
Suppress warnings "mesh_use_embedded_materials is ignored"

rviz more stricly checks marker parameters and thus issues this warning
if mesh_use_embedded_materials was set although no mesh_resource was specified

bostoncleek committed:
Improve robustness of subframes test (#2488)

martiniil committed:
Use kinematics solver timeout if not specified otherwise (#2489)

Commit:
Do not break out of loop -- need to update low pass filters (#2496)

Commit:
Protect paused_ flag, for thread safety (#2494)

ct2034 committed:
Add missing dependency on joint_limits_interface (#2487)

Commit:
Refactor Servo velocity bounds enforcement (#2471)

* Refactor/fix Servo enforceVelLimits function to utilize joint velocity limits correctly.

* Use uniform scaling factor instead of joint specific velocity bounds.

* Move joint increment outside of if()

* Add friend test for velocity limits

* Clean up debug code, clang format

* Test for negative velocities, too

* Apply velocity scaling correctly and skip clamping zero velocities.

* Improve enforceVelLimits unit tests.

* Use EXPECT_NEAR in the new unit test

Co-authored-by: AndyZe <andyz@utexas.edu>

pawanw17 committed:
resolved typo in first-timers-only.md (#2443)

Commit:
Fix logic, improve function comment for clearDiffs() (#2497)

Commit:
start adding collision temrs

Commit:
changed header extension

Commit:
adjusted CalcCollisions and CollisionstoDistance and CollisionsToDistanceExpressions

Commit:
no tesseract but the functionality should be tested

Commit:
clean

Commit:
replace GetKin() with GetPlanningScene

Commit:
comment

Commit:
added a conversion between Contact from MoveIt and ContactResult from tesseract

Commit:
test

Commit:
use PlanningScene

ommmid committed:
implemented planning scene for collision term too. It seems to be correct. Just used it with FCL and singletimestep not continous collision checking yet.

ommmid committed:
woo-hoo! Now CollisionTermInfo of trajopt is implemented.

This version successfully uses Bullet collision detector from MoveIt to do the collsion checking and find singed distances. There are a lot of changes in different
files but the most changes are done in collision_terms.cpp

Things left to do:
1. This implementation is hard-coded. But it is easy to change this to a version that the user can enter a CollisionTermInfo.
Currently this is done in trajopt_interface.cpp but it should be defined by user, its parametes should be added to setup.yaml as well.

2. It needs cleanup in the code base.

3. The PlanningScene that motion planners have acces to is a ConsPtr but we need to make changes in RobotState to be able to check collision. To this end,
a child of the original PlanningScene is used through diff() function. For some reason, I can not use "checkCollision" function on this planning scene child.
It turned out if I remove all the objects from the child and add them again, "checkCollision" operates correctly
